h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,204 @@
+package hex.genmodel.tools;
+
+import com.mxgraph.layout.mxCompactTreeLayout;
+import com.mxgraph.layout.mxEdgeLabelLayout;
+import com.mxgraph.layout.mxIGraphLayout;
+import com.mxgraph.util.mxCellRenderer;
+import hex.genmodel.algos.tree.ConvertTreeOptions;
+import hex.genmodel.algos.tree.SharedTreeGraph;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.genmodel.algos.tree.TreeBackedMojoModel;
+import org.jgrapht.ext.JGraphXAdapter;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.DirectedMultigraph;
+import org.jgrapht.io.*;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    @Override
+    public void run() throws Exception {
+        if (!Format.png.equals(format)){
+            super.run();
+        } else {
+            validateArgs();
+            if (genModel instanceof TreeBackedMojoModel){
+                TreeBackedMojoModel treeBackedModel = (TreeBackedMojoModel) genModel;
+                ConvertTreeOptions options = new ConvertTreeOptions().withTreeConsistencyCheckEnabled();
+                final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+                printPng(g);
+            }
+            else {
+                System.out.println("ERROR: Unknown MOJO type");
+                System.exit(1);
+            }
+        } 
+    }
+
+    @Override
+    public boolean supportsFormat(Format format){
+        return true;
+    }
+    
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        Path outputDirectoryPath = Paths.get(outputFileName);
+        int numberOfTrees = trees.subgraphArray.size();
+        if (numberOfTrees > 1) { 
+            if (outputFileName == null) {
+                outputFileName = Paths.get("").toString();
+            }
+            if (Files.exists(outputDirectoryPath) && !Files.isDirectory(outputDirectoryPath)) {
+                Files.delete(outputDirectoryPath);
+            }
+            if (!Files.exists(outputDirectoryPath)) {
```

Comment:
also if the output needs to be a directory and it is not, we should just fail immediately with an error

Reply:
hmm, this might necesarily not be true, if the output is a regular file than we can ovewrite it with a directory, so this would be reasonable to do